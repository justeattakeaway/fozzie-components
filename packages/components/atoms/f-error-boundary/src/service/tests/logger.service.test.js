import loggerService from '../logger.service';

const error = {
    name: 'Error adding to basket',
    message: 'timeout of 5000ms exceeded.',
    stack: 'at XMLHttpRequest.ontimeout (bundle.js:12:345)',
    toString: () => 'Error adding to basket'
};

const info = 'mounted hook';

describe('loggerService', () => {
    describe('buildVueError', () => {
        describe('when tier is passed', () => {
            it('should return expected object', () => {
                // Arrange
                const vm = undefined;
                const tier = '1';

                // Act
                const result = loggerService.buildVueError(error, vm, info, { tier });

                // Assert
                expect(result).toMatchSnapshot();
            });
        });

        describe('when vm is undefined', () => {
            it('should return expected object', () => {
                // Arrange
                const vm = undefined;

                // Act
                const result = loggerService.buildVueError(error, vm, info);

                // Assert
                expect(result).toMatchSnapshot();
            });
        });

        describe('when vm has props data', () => {
            it('should return expected object', () => {
                // Arrange
                const vm = {
                    $options: {
                        propsData: {
                            name: 'basket'
                        }
                    }
                };

                // Act
                const result = loggerService.buildVueError(error, vm, info);

                // Assert
                expect(result).toMatchSnapshot();
            });
        });

        describe('when vm is for root component', () => {
            it('should return expected object', () => {
                // Arrange
                const vm = {};

                vm.$root = vm;

                // Act
                const result = loggerService.buildVueError(error, vm, info);

                // Assert
                expect(result).toMatchSnapshot();
            });
        });

        describe('when vm contains component name', () => {
            it('should return expected object', () => {
                // Arrange
                const vm = {
                    name: 'Basket Component'
                };

                // Act
                const result = loggerService.buildVueError(error, vm, info);

                // Assert
                expect(result).toMatchSnapshot();
            });
        });

        describe('when vm options contains name and file name', () => {
            it('should return expected object', () => {
                // Arrange
                const vm = {
                    _isVue: true,
                    $options: {
                        name: 'Basket Component',
                        __file: 'BasketComponent.vue'
                    }
                };

                // Act
                const result = loggerService.buildVueError(error, vm, info);

                // Assert
                expect(result).toMatchSnapshot();
            });
        });

        describe('when vm options contains name', () => {
            it('should return expected object', () => {
                // Arrange
                const vm = {
                    _isVue: true,
                    $options: {
                        name: 'Basket Component'
                    }
                };

                // Act
                const result = loggerService.buildVueError(error, vm, info);

                // Assert
                expect(result).toMatchSnapshot();
            });
        });

        describe('when vm options contains component tag', () => {
            it('should return expected object', () => {
                // Arrange
                const vm = {
                    _isVue: true,
                    $options: {
                        _componentTag: 'Basket Component'
                    }
                };

                // Act
                const result = loggerService.buildVueError(error, vm, info);

                // Assert
                expect(result).toMatchSnapshot();
            });
        });
    });
});
