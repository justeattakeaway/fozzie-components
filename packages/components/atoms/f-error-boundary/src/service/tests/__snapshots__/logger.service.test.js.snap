// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loggerService buildVueError when tier is passed should return expected object 1`] = `
Object {
  "Exception": "Error adding to basket",
  "ExceptionMessage": "timeout of 5000ms exceeded.",
  "ExceptionStackTrace": "at XMLHttpRequest.ontimeout (bundle.js:12:345)",
  "Tier": "1",
  "vue.componentName": "vm not found",
  "vue.lifecycleHook": "mounted hook",
  "vue.propsData": undefined,
}
`;

exports[`loggerService buildVueError when vm contains component name should return expected object 1`] = `
Object {
  "Exception": "Error adding to basket",
  "ExceptionMessage": "timeout of 5000ms exceeded.",
  "ExceptionStackTrace": "at XMLHttpRequest.ontimeout (bundle.js:12:345)",
  "vue.componentName": "component <Basket Component>",
  "vue.lifecycleHook": "mounted hook",
  "vue.propsData": undefined,
}
`;

exports[`loggerService buildVueError when vm has props data should return expected object 1`] = `
Object {
  "Exception": "Error adding to basket",
  "ExceptionMessage": "timeout of 5000ms exceeded.",
  "ExceptionStackTrace": "at XMLHttpRequest.ontimeout (bundle.js:12:345)",
  "vue.componentName": "anonymous component",
  "vue.lifecycleHook": "mounted hook",
  "vue.propsData": Object {
    "name": "basket",
  },
}
`;

exports[`loggerService buildVueError when vm is for root component should return expected object 1`] = `
Object {
  "Exception": "Error adding to basket",
  "ExceptionMessage": "timeout of 5000ms exceeded.",
  "ExceptionStackTrace": "at XMLHttpRequest.ontimeout (bundle.js:12:345)",
  "vue.componentName": "root instance",
  "vue.lifecycleHook": "mounted hook",
  "vue.propsData": undefined,
}
`;

exports[`loggerService buildVueError when vm is undefined should return expected object 1`] = `
Object {
  "Exception": "Error adding to basket",
  "ExceptionMessage": "timeout of 5000ms exceeded.",
  "ExceptionStackTrace": "at XMLHttpRequest.ontimeout (bundle.js:12:345)",
  "vue.componentName": "vm not found",
  "vue.lifecycleHook": "mounted hook",
  "vue.propsData": undefined,
}
`;

exports[`loggerService buildVueError when vm options contains component tag should return expected object 1`] = `
Object {
  "Exception": "Error adding to basket",
  "ExceptionMessage": "timeout of 5000ms exceeded.",
  "ExceptionStackTrace": "at XMLHttpRequest.ontimeout (bundle.js:12:345)",
  "vue.componentName": "component <Basket Component>",
  "vue.lifecycleHook": "mounted hook",
  "vue.propsData": undefined,
}
`;

exports[`loggerService buildVueError when vm options contains name and file name should return expected object 1`] = `
Object {
  "Exception": "Error adding to basket",
  "ExceptionMessage": "timeout of 5000ms exceeded.",
  "ExceptionStackTrace": "at XMLHttpRequest.ontimeout (bundle.js:12:345)",
  "vue.componentName": "component <Basket Component> at BasketComponent.vue",
  "vue.lifecycleHook": "mounted hook",
  "vue.propsData": undefined,
}
`;

exports[`loggerService buildVueError when vm options contains name should return expected object 1`] = `
Object {
  "Exception": "Error adding to basket",
  "ExceptionMessage": "timeout of 5000ms exceeded.",
  "ExceptionStackTrace": "at XMLHttpRequest.ontimeout (bundle.js:12:345)",
  "vue.componentName": "component <Basket Component>",
  "vue.lifecycleHook": "mounted hook",
  "vue.propsData": undefined,
}
`;
